src/FGS.Linq.Extensions.OneOf.EntityFramework.Shared/OneOfQueryableExtensions.cs

```diff
@@ -27,7 +27,7 @@ public static class OneOfQueryableExtensions
         /// <returns>A task that represents the asynchronous operation. The task result contains the single element from the input query, or the reason why it could be returned.</returns>
         public static async Task<OneOf<T, NoElements, MoreThanOneElement>> MaterializeSingleOrReasonWhyNot<T>(this IQueryable<T> source, CancellationToken cancellationToken = default)
         {
-            var firstOrDefault = (await source.Take(1).MaterializeAsync(cancellationToken).ConfigureAwait(false)).First();
+            var firstOrDefault = (await source.Take(1).MaterializeAsync(cancellationToken).ConfigureAwait(false)).MaterializeFirstOrDefaultAsync();
```

Comment:
This now has two materializations, that's not right is it?

Reply:
No, no it is not. :-(